tools/deprecate_features/deprecate_features.py

```diff
@@ -30,13 +30,18 @@ def deprecate_proto():
         filenames_and_fields.discard(tuple([match.group(1), match.group(2)]))
 
   # Finally sort out the code to add to runtime_features.cc and a canned email for envoy-announce.
-  code = ''
-  email = '\nThe following deprecated configuration fields will be disallowed by default:\n'
+  code_snippets = []
+  email_snippets = []
   for (filename, field) in filenames_and_fields:
-    code += ('    "envoy.deprecated_features.' + filename + ':' + field + '",\n')
-    email += (field + ' from ' + filename + '\n')
+    code_snippets.append('    "envoy.deprecated_features.' + filename + ':' + field + '",\n')
+    email_snippets.append(field + ' from ' + filename + '\n')
+  code = ''.join(code_snippets)
+  email = ''
+  if email_snippets:
+    email = ('\nThe following deprecated configuration fields will be disallowed by default:\n' +
+             ''.join(email_snippets))
```

Comment:
Looks good, but one slight Pythonic nit. Could we do `'\n'.join(email_snippets)'`? That way the above `append` becomes a bit more declarative. Same for `code_snippets` (and below..).

Reply:
I can, but I'd prefer not to if I can talk you out of it :-)

If I pull newlines out of email snippets I think I should do it for the code for consistency.  But for code it's a need-to-have and for email it's a nice-to-have to have the trailing \n after the join.  I mean I could append another newline after both but if I want it after each snippet rather than between each snippet I think having it in the snippet rather than having it in the join is semantically correct.

Reply:
It's semantically correct, I'm just suggesting there are more Pythonic ways to do it if you're after Python style suggestions. Since the join value is substituted into a string in both cases, you can write something like `'foo bar:\n %s\n' % '\n'.join(stuff without any new lines in it)`. This discussion is really me being nit-picky, but reflects my understanding of Python best practices.